### PR TITLE
ccacheStdenv: prefer default extraConfig from ccacheWrapper

### PR DESCRIPTION
**Pull request by kira-bruneau**
###### Motivation for this change
This commit brings back the original behaviour before commit 82ca111efba63590c65e23019db1c1a6bb7f4bbd where `ccacheStdenv` would use the `extraConfig` from `ccacheWrapper` when `ccacheWrapper` was overridden.

See https://github.com/NixOS/nixpkgs/issues/104425#issuecomment-755157686

###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

@r-burns @ElXreno

### CODE REVIEW

**Comment by MetaDark** [comment]
Result of `nixpkgs-review pr 108594` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -11364,10 +11364,11 @@ in
 
   ccacheStdenv = lowPrio (makeOverridable ({ extraConfig, stdenv }:
     overrideCC stdenv (buildPackages.ccacheWrapper.override {
-      inherit extraConfig;
       inherit (stdenv) cc;
+    } // lib.optionalAttrs (extraConfig != null) {
```
Comment (ElXreno) [inline pkgs/top-level/all-packages.nix:11368]: In my opinion, `null' is not equivalent to an empty string. I think there may be some problems here.
How about comparing it to an empty string? Or is this not required and everything will work?

Comment (MetaDark) [reply pkgs/top-level/all-packages.nix:11368]: I changed the default value of `extraConfig` for `ccacheStdenv` from an empty string to null to use as a marker to indicate that the field hasn't been overwritten, so there shouldn't be any problem with this check.

Although, this means that if null is `extraConfig` is explicitly overridden with null, the default `extraConfig` will still be used from `ccacheWrapper` which may be confusing. I will try to make this work without a null marker.

**Review by r-burns — APPROVED** [review]

**Comment by r-burns** [comment]
4.2.1 test.multi_arch fails on darwin, with:

```
/tmp/nix-build-ccache-4.2.1.drv-0/source/test/run: line 342: compgen: command not found
Running test suite multi_arch/tmp/nix-build-ccache-4.2.1.drv-0/source/test/run: line 342: compgen: command not found
.
FAILED

Test suite:     multi_arch (line 20)
Test case:      cache hit, direct mode
Failure reason: Expected "cache miss" to be 1, actual 0
```

**Comment by MetaDark** [comment]
Oh sorry @r-burns, I didn't mean to update this PR, I pushed to my ccache branch to try out the build in a Mac VM. I'll try to see if I can fix this failure without disabling the test.

**Pull request closed without merging**